docs/proposals/proposal-audit-log.md

```diff
@@ -0,0 +1,209 @@
+# Audit log
+
+### Purpose of audit log
+
+The audit log is a feature that allows users to track changes made to their organization, projects,
+schemas, and other resources.
+
+## Events to track
+
+### OrganizationManager
+
+- `leaveOrganization`
+- `createOrganization`
+- `deleteOrganization`
+- `updatePlan`
+- `updateRateLimits`
+- `updateName`
+- `deleteInvitation`
+- `inviteByEmail`
+- `joinOrganization`
+- `requestOwnershipTransfer`
+- `answerOwnershipTransferRequest`
+- `deleteMember`
+- `updateMemberAccess`
+- `createMemberRole`
+- `deleteMemberRole`
+- `assignMemberRole`
+- `updateMemberRole`
+- `createRoleWithMembersMigration`
+- `assignRoleToMembersMigration`
+
+  ### ProjectManager
+
+- `createProject`
+- `deleteProject`
+- `updateName`
+
+### ArtifactStorageWriter
+
+- `writeArtifact`
+- `deleteArtifact`
+
+### CompositeLegacyModel
+
+- `check`
+- `publish`
+
+### SingleLegacyModel
+
+- `check`
+- `publish`
```

Comment:
these are invoked via SchemaPublisher and other classes, no need to track them individually.